File: Codes/3152-special-array-ii.cpp
/*
 * @Author: LetMeFly
 * @Date: 2024-08-14 23:26:29
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-08-14 23:33:15
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    vector<bool> isArraySpecial(vector<int>& nums, vector<vector<int>>& queries) {
        int cnt = 0;
        for (int i = 0; i < nums.size() - 1; i++) {
            bool same = nums[i] % 2 == nums[i + 1] % 2;
            nums[i] = cnt;
            cnt += same;
        }
        nums.back() = cnt;
        vector<bool> ans(queries.size());
        for (int i = 0; i < queries.size(); i++) {
            ans[i] = nums[queries[i][1]] == nums[queries[i][0]];
        }
        return ans;
    }
};

File: Codes/3152-special-array-ii.java
/*
 * @Author: LetMeFly
 * @Date: 2024-08-14 23:36:10
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-08-14 23:38:18
 */
class Solution {
    public boolean[] isArraySpecial(int[] nums, int[][] queries) {
        int cnt = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            boolean same = nums[i] % 2 == nums[i + 1] % 2;
            nums[i] = cnt;
            cnt += same ? 1 : 0;
        }
        nums[nums.length - 1] = cnt;
        boolean[] ans = new boolean[queries.length];
        for (int i = 0; i < queries.length; i++) {
            ans[i] = nums[queries[i][0]] == nums[queries[i][1]];
        }
        return ans;
    }
}